packages/cozy-doctypes/src/Folder.js

```diff
@@ -0,0 +1,105 @@
+const Application = require('./Application')
+const CozyFile = require('./File')
+
+/**
+ * Class representing the folder model.
+ * @extends CozyFile
+ */
+class CozyFolder extends CozyFile {
+  /**
+   * Create a folder with a reference to the given document
+   * @param  {String}  path     Folder path
+   * @param  {Object}  document
+   * @return {Object}  Folder document
+   */
+  static async createFolderWithReference(path, document) {
+    const collection = this.cozyClient.collection(CozyFile.doctype)
+    const dirId = await collection.ensureDirectoryExists(path)
+    await collection.addReferencesTo(document, [
+      {
+        _id: dirId
+      }
+    ])
+
+    const { data: dirInfos } = await collection.get(dirId)
+
+    return dirInfos
+  }
+
+  /**
+   * Returns a "Magic Folder", given its id
+   * @param  {String} id Magic Folder id. Could be:
+   * * `CozyFolder.refs.ADMINISTRATIVE`
+   * * `CozyFolder.refs.PHOTOS`
+   * * `CozyFolder.refs.PHOTOS_BACKUP`
+   * * `CozyFolder.refs.PHOTOS_UPLOAD`
```

Comment:
I think having the list of folders here is not good. It will be out of date since someone will update the CozyFolder.refs and not update this list.

```suggestion
   * CozyFolder.refs contains the ids of folders that can be magic folders.
```

Writing this comment, I was thinking if we'd better not rename `CozyFolder.refs` into `CozyFolders.magicFolders`. It will be clearer and we'll have less risk of someone wanting to store a ref to a document there, *without* wanting it to be a magic folder.

Reply:
I was thinking the same thing, let's go.